front/src/modules/trainschedule/components/Timetable/Timetable.tsx

```diff
@@ -96,6 +98,20 @@ const Timetable = ({
     }
   };
 
+  const currentDepartureDates = useMemo(
+    () => displayedTrainSchedules.map((train) => formatDepartureDate(train.startTime)),
+    [displayedTrainSchedules]
+  );
+
+  const showDepartureDates = useMemo(() => {
+    let previousDepartureDate = '';
+    return currentDepartureDates.map((date, index) => {
+      const show = index === 0 || date !== previousDepartureDate;
```

Comment:
Nit: the `index === 0` condition should be unnecessary, since initially `previousDepartureDate = ''` so should always be different from any formatted date.

Reply:
done